src/include/migraphx/op/scatter.hpp

```diff
@@ -37,25 +38,28 @@ struct scatter
 
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3).standard();
+        check_shapes{inputs, *this}.has(3);
+        check_shapes{{inputs.at(1), inputs.at(2)}, *this}.same_dims();
         return inputs.front();
     }
 
     argument compute(const shape& output_shape, std::vector<argument> args) const
     {
         argument result{output_shape};
+        const shape s_output_shape{output_shape.type(), output_shape.lens()};
         // max dimension in axis
         auto axis_dim_size = output_shape.lens()[axis];
         visit_all(result, args[0], args[2])([&](auto output, auto data, auto update) {
             std::copy(data.begin(), data.end(), output.begin());
             args[1].visit([&](auto indices) {
```

Comment:
the std::copy() will not work, either since data can be nonstd shape, while output will always be standard.

Reply:
Do something like:

```
            shape_for_each(output_shape, [&](const auto& idx) {
                output(idx.begin(), idx.end()) = data(idx.begin(), idx.end());
            });
```